halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/openstack/OpenstackAccountValidator.java

```diff
@@ -76,19 +112,27 @@ public void validate(ConfigProblemSetBuilder psBuilder, OpenstackAccount account
     lbaasConfig.setPollTimeout(lbaas.getPollTimeout());
 
     try {
-      List<String> regionsList = Arrays.asList(StringUtils.split(regions, ","));
       OpenstackNamedAccountCredentials openstackCredentials = new OpenstackNamedAccountCredentials.Builder()
           .name(account.getName())
+          .environment(environment)
+          .accountType(accountType)
           .authUrl(authUrl)
           .username(username)
           .password(password)
           .projectName(projectName)
           .domainName(domainName)
-          .regions(regionsList)
+          .regions(regions)
           .insecure(insecure)
+          .heatTemplateLocation(heatTemplateLocation)
+          .consulConfig(consulConfig)
           .lbaasConfig(lbaasConfig)
           .userDataFile(userDataFile)
           .build();
+      if (openstackCredentials == null) {
+        return;
```

Comment:
I'm fairly certain that it's impossible for this value to be null at this point: Java constructors cannot return null, and the build() method only calls a constructor and returns its result.

Did this check come about in response to an error? Or does it simply conform to convention elsewhere within the spinnaker/halyard code base? I'm new here so this might be a silly question. @edwinavalos